tests/www/apply/test_process_external_transfer.py

```diff
@@ -0,0 +1,517 @@
+    response = client.get(transfer_step_2_url)
+
+    assert str(parse_response_to_soup(response, ".c-stepper")) == snapshot(name="progress")
+    assertContains(response, "<h2>Sélectionner les métiers recherchés</h2>", html=True)
+    assert response.context["form"].initial == {"selected_jobs": [], "spontaneous_application": True}
+
+    response = client.post(transfer_step_2_url, data={"spontaneous_application": "on"})
+    transfer_step_3_base_url = reverse(
+        "apply:job_application_external_transfer_step_3",
+        kwargs={"job_application_id": job_application.pk, "company_pk": other_company.pk},
+    )
+    transfer_step_3_url = f"{transfer_step_3_base_url}?back_url={quote(transfer_step_2_url)}"
+    assertRedirects(response, transfer_step_3_url)
+    assert client.session[f"job_application-{other_company.pk}"] == {"selected_jobs": []}
+
+    # With selected job
+    transfer_step_2_url = (
+        f"{transfer_step_2_base_url}?job_description_id={job_id}&back_url={quote(transfer_step_1_url)}"
+    )
+    response = client.get(transfer_step_2_url)
+
+    assert str(parse_response_to_soup(response, ".c-stepper")) == snapshot(name="progress")
+    assertContains(response, "<h2>Sélectionner les métiers recherchés</h2>", html=True)
+    assert response.context["form"].initial == {"selected_jobs": [str(job_id)]}
+
+    response = client.post(transfer_step_2_url, data={"selected_jobs": [job_id]})
+    transfer_step_3_url = f"{transfer_step_3_base_url}?back_url={quote(transfer_step_2_url)}"
+    assertRedirects(response, transfer_step_3_url)
+    assert client.session[f"job_application-{other_company.pk}"] == {"selected_jobs": [job_id]}
+
+
+@freeze_time("2024-07-15 11:52:23")
+def test_step_3(client, snapshot):
+    job_application = JobApplicationFactory(state=JobApplicationState.REFUSED, for_snapshot=True)
+    employer = job_application.to_company.members.get()
+    other_company = CompanyFactory(with_membership=True)
+    client.force_login(employer)
+
+    transfer_step_2_url = reverse(
+        "apply:job_application_external_transfer_step_2",
+        kwargs={"job_application_id": job_application.pk, "company_pk": other_company.pk},
+    )
+    transfer_step_3_base_url = reverse(
+        "apply:job_application_external_transfer_step_3",
+        kwargs={"job_application_id": job_application.pk, "company_pk": other_company.pk},
+    )
+    transfer_step_3_url = f"{transfer_step_3_base_url}?back_url={quote(transfer_step_2_url)}"
+    response = client.get(transfer_step_3_url)
+
+    assert str(parse_response_to_soup(response, ".c-stepper")) == snapshot(name="progress")
+    expected_message = f"Le 15/07/2024 à 11h52, Pierre DUPONT a écrit :\n\n{job_application.message}"
+    assert response.context["form"].initial["message"] == expected_message
+
+    response = client.post(transfer_step_3_url, data={"message": expected_message, "keep_original_resume": True})
```

Comment:
On ne peut pas envoyer de booléen en HTTP. (multiples endroits)
```suggestion
    response = client.post(transfer_step_3_url, data={"message": expected_message, "keep_original_resume": "True"})
```